tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,460 @@
+	externalIPAddresses []string) []*types.NetworkSelectionElement {
+	ipAnnotation := pod.StaticIPAnnotation(internalNADName, internalIPAddresses)
+
+	return append(ipAnnotation,
+		pod.StaticIPAnnotation(externalNADName, externalIPAddresses)...)
+}
+
+func createHubConfigMap(name string) *configmap.Builder {
+	configMapData := defineBaseConfig(tsparams.DaemonsFile, "", "")
+	hubConfigMap, err := configmap.NewBuilder(APIClient, name, tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create hub config map")
+
+	return hubConfigMap
+}
+
+func createMasterConfigMap(name, configMapStaticRoutes string) *configmap.Builder {
+	configMapData := defineBaseConfig(tsparams.DaemonsFile, configMapStaticRoutes, "")
+	masterConfigMap, err := configmap.NewBuilder(APIClient, name, tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create master config map")
+
+	return masterConfigMap
+}
+
+func createFrrPodTest(name, nodeName, configmapName, containerName string, defaultCMD []string,
+	secondaryNetConfig []*types.NetworkSelectionElement, masterPod bool) *pod.Builder {
+	var (
+		frrCtr       *corev1.Container
+		frrContainer *pod.ContainerBuilder
+		err          error
+	)
+
+	By("Creating FRR master container in the test namespace")
+
+	if masterPod {
+		frrContainer = pod.NewContainerBuilder(
+			containerName, NetConfig.CnfNetTestContainer, []string{"/bin/bash", "-c",
+				"testcmd -interface net1 -protocol tcp -port 8888 -listen"})
+	} else {
+		frrContainer = pod.NewContainerBuilder(
+			containerName, NetConfig.CnfNetTestContainer, []string{"/bin/bash", "-c", "sleep INF"})
+	}
+
+	frrContainer.WithSecurityCapabilities([]string{"NET_ADMIN", "NET_RAW", "SYS_ADMIN"}, true)
+	frrCtr, err = frrContainer.GetContainerCfg()
+	Expect(err).ToNot(HaveOccurred(), "Failed to get container configuration")
+
+	frrPod, err := pod.NewBuilder(APIClient, name, tsparams.TestNamespaceName, NetConfig.FrrImage).
+		DefineOnNode(nodeName).
+		WithTolerationToMaster().
+		WithSecondaryNetwork(secondaryNetConfig).
+		RedefineDefaultCMD(defaultCMD).
+		WithAdditionalContainer(frrCtr).
+		WithLocalVolume(configmapName, "/etc/frr").
+		WithPrivilegedFlag().
+		CreateAndWaitUntilRunning(5 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create FRR test pod")
+
+	return frrPod
+}
+
+func setWorkerNodeListAndLabelForTests(workerNodeList []*nodes.Builder) (map[string]string, []*nodes.Builder) {
+	return NetConfig.WorkerLabelMap, workerNodeList
+}
+
+func defineBaseConfig(daemonsConfig, frrConfig, vtyShConfig string) map[string]string {
+	configMapData := make(map[string]string)
+	configMapData["daemons"] = daemonsConfig
+	configMapData["frr.conf"] = frrConfig
+	configMapData["vtysh.conf"] = vtyShConfig
+
+	return configMapData
+}
+
+func removePrefixFromIPList(ipAddressList []string) []string {
```

Comment:
reuse from common

Reply:
that is in metallb should I move it upward?

Reply:
moved to network.internal.frrconfig